core/src/main/scala/chisel3/Intrinsic.scala

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3
+
+import chisel3._
+import chisel3.experimental.{requireIsChiselType, Param, SourceInfo}
+import chisel3.internal.firrtl.ir._
+import chisel3.internal.{Builder, OpBinding}
+import chisel3.internal.Builder.pushCommand
+
+object Intrinsic {
+
+  /**
+    * Create an intrinsic statement.
+    *
+    * @param intrinsic name of the intrinsic
+    * @param params parameters, if any
+    * @param data inputs
+    *
+    * @example {{{
+    * Intrinsic("test", Map("Foo" -> 5))(f, g)
+    * }}}
+    */
+  def apply(intrinsic: String, params: Map[String, Param])(data: Data*)(implicit sourceInfo: SourceInfo): Unit = {
```

Comment:
Hmm, FWIW this is the same as the BlackBox / IntrinsicModule API -- and the params are sorted by key as part of the conversion (`params.keys.toList.sorted.map { name => convert(name, params(name)) }` so at least it's deterministic.  That doesn't mean those are "right" either but just so we're looking at the same things 👍 .

I'm happy to use any of your suggestions although unordered map on API kinda makes sense to me if we're going to sort anyway (and Map ensures one value per key, otherwise the last two suggestions (~ `Seq[(String, Param)]`) seem best to me.

WDYT?

Reply:
I missed the sorting first time, thanks for pointing it out!

We made a mistake on the BlackBox API using Map and the sorting was the best workaround we could come up with without making it source incompatible (and it didn't feel worth it to break code). IMO it's better to preserve user order but if you think sorting is better that's also fine.

Reply:
Sounds good!  Preserving user-order feels nice, agreed, although I'm not sure there are any guarantees they'll reach the SV that way presently. 

So, I guess `VectorMap[String, Param]` ("OrderedMap" 😅 ) on API or I think the idea is to not prefer specifying storage on interfaces, so the seq version (/variadic) with requirement that keys are unique?

Thanks for the attention and review! 👍

Reply:
Went with variadic + require after all, thanks!